scripts/jenkins/ardana/gerrit/gerrit_handle_event.py

```diff
@@ -0,0 +1,166 @@
+#!/usr/bin/env python
+
+import argparse
+import os
+import sys
+
+sys.path.append(os.path.dirname(__file__))
+from gerrit import GerritChange, GerritChangeSet  # noqa: E402
+from gerrit_merge import gerrit_merge  # noqa: E402
+from gerrit_review import gerrit_review  # noqa: E402
+
+
+def get_submittable_references(merged_change):
+    """
+    Get all submittable changes that directly depend on the supplied change
+    (have an explicit Depends-On referencing the indicated merged Gerrit
+    change and targeting the same branch, or that are submitted together
+    and directly on top of it).
+
+    :param merged_change: GerritChange object representing a merged change
+    :return: list of GerritChange object representing submittable changes,
+    referencing the supplied merged change as a direct dependency, and which
+    could be merged themselves
+    """
+
+    changeset = GerritChangeSet(
+        'is:open',
+        'is:mergeable',
+        'branch:{}'.format(merged_change.branch),
+        'label:Code-Review=2',
+        'label:Verified=2',
+        'label:Workflow=1',
+        '-label:Verified=-1',
+        '-label:Code-Review=-2'
+    )
+
+    references = []
+    for change in changeset.changes():
+        if change.submittable and change.depends_on_directly(merged_change):
+            references.append(change)
+
+    return references
+
+
+def get_stale_references(updated_change):
+    """
+    Get all changes that directly or indirectly (i.e. through a chain of
+    Depends-On commit message indicators) depend on the indicated updated
+    Gerrit change and targeting the same branch.
```

Comment:
Note, this method only creates a list of stale depends-on references. It does not return any child commits that may be dependant on the given updated_change.

I'm not sure if we'll want to be rechecking the children though since if the uploader did not re-upload them themselves, then they would need rebasing.

Reply:
I have also updated the `depends_on_directly` method so that this one will also include child commits, but you have a point that those will need rebasing. I need to test this scenario and see if I can detect this "needs rebase" condition in the CI, to avoid triggering builds unnecessarily. 
This also makes me thing we should abort the Gerrit CI early if the target change or any of its dependencies require rebasing or have conflicts with the master branch.

More to do for me...

Reply:
Actually, scratch that, I used a different approach: there is a "get related changes" Gerrit API call that can help me to avoid abusing the `_get_uploaded_together` call unnecessarily. I implemented a `get_related` method which uses that, which then I use everywhere where I need to check references.

Reply:
I might be missing something, but I can't see this new method sorry?

Reply:
sorry, the method is `_get_related_changes`. I also had to remove `_get_uploaded_together` because it was unreliable and slightly overlapped with `_get_related_changes`.